provider/boltmem/boltmem.go

```diff
@@ -261,15 +261,15 @@ func (s *Silences) Mutes(lset model.LabelSet) bool {
 
 // All returns all existing silences.
 func (s *Silences) All() (*types.SilencesQueryResponse, error) {
-	return s.Query(len(s.k), 0)
+	return s.Query(len(s.sils), 0, types.ByCreatedAt)
 }
 
 // Query returns n silences starting at page offset o.
-func (s *Silences) Query(n, o int) (*types.SilencesQueryResponse, error) {
+func (s *Silences) Query(n, o int, fn types.SilencesLessFunc) (*types.SilencesQueryResponse, error) {
```

Comment:
Can be a follow-up commit, but `uint` would be the better type here. n/o are not descriptive variable names to be used in a exported function signature, I'd suggest `limit, offset uint, ...`